test/NuGet.Core.Tests/NuGet.Build.Tasks.Test/GetReferenceNearestTargetFrameworkTaskTest.cs

```diff
@@ -469,5 +469,161 @@ public void GetReferenceNearestTargetFrameworkTask_WithSingleTfmAndSingleAtfAndN
+        [InlineData(".NETCoreApp,Version=v6.0", "Android,Version=22.0", "net5.0;net5.0-android", ".NETCoreApp,Version=v5.0;.NETCoreApp,Version=v5.0", "None;android,Version=21.0", "net5.0-android")]
+        [InlineData(".NETCoreApp,Version=v5.0", null, "latestnet;latestnetstandard", ".NETFramework,Version=v4.7.2;.NETStandard,Version=v2.1", "None;None", "latestnetstandard")]
+        public void GetReferenceNearestTargetFrameworkTask_WithTargetFrameworkInformation_ReturnsCompatibleAlias(
+            string currentProjectTFM, string currentProjectTPM, string refTargetFrameworks, string refTargetFrameworkMonikers, string refTargetPlatformMonikers, string expected)
+        {
+            var buildEngine = new TestBuildEngine();
+            var testLogger = buildEngine.TestLogger;
+
+            var references = new List<ITaskItem>();
+            var reference = new Mock<ITaskItem>();
+            reference.SetupGet(e => e.ItemSpec).Returns("a.csproj");
+            reference.Setup(e => e.GetMetadata("TargetFrameworks")).Returns(refTargetFrameworks);
+            reference.Setup(e => e.GetMetadata("TargetFrameworkMonikers")).Returns(refTargetFrameworkMonikers);
+            reference.Setup(e => e.GetMetadata("TargetPlatformMonikers")).Returns(refTargetPlatformMonikers);
+            references.Add(reference.Object);
+
+            var task = new GetReferenceNearestTargetFrameworkTask
+            {
+                BuildEngine = buildEngine,
+                CurrentProjectTargetFramework = currentProjectTFM,
+                CurrentProjectTargetPlatform = currentProjectTPM,
+                FallbackTargetFrameworks = new string[] { },
+                AnnotatedProjectReferences = references.ToArray()
+            };
+
+            var result = task.Execute();
+            result.Should().BeTrue(because: testLogger.ShowMessages());
+
+            task.AssignedProjects.Should().HaveCount(1);
+            task.AssignedProjects[0].GetMetadata("NearestTargetFramework").Should().Be(expected);
+
+            testLogger.Warnings.Should().Be(0);
+            testLogger.Errors.Should().Be(0);
+            testLogger.DebugMessages.Count.Should().Be(DEBUG_MESSAGE_COUNT_INPUT_OUTPUT);
+        }
+
+        [Theory]
+        [InlineData(".NETFramework,Version=v4.7.2", null, "net46;net472", ".NETFramework,Version=v4.6", "None")]
+        [InlineData(".NETFramework,Version=v4.7.2", null, "net46", ".NETFramework,Version=v4.6;.NETFramework,Version=v4.7.2", "None")]
+        [InlineData(".NETFramework,Version=v4.7.2", null, "net46;net472", ".NETFramework,Version=v4.6", "None;Windows,Version=77.0")]
+        public void GetReferenceNearestTargetFrameworkTask_WithInvalidParameters_Errors(
+    string currentProjectTFM, string currentProjectTPM, string refTargetFrameworks, string refTargetFrameworkMonikers, string refTargetPlatformMonikers)
```

Comment:
I'd add a case to this test for `net5.0` trying to reference `net5.0-windows` or `net5.0-windows;net5.0-androad`.